packages/sdk/src/fee/FeeEstimator.ts

```diff
@@ -0,0 +1,144 @@
+import { Psbt } from "bitcoinjs-lib"
+
+import { AddressFormats, getNetwork, getScriptType } from ".."
+import { Network } from "../config/types"
+import { MAXIMUM_FEE } from "../constants"
+import { FeeEstimatorOptions } from "./types"
+
+export default class FeeEstimator {
+  feeRate: number
+  network: Network
+  psbt!: Psbt
+  witness?: Buffer[] = []
+  fee = 0
+
+  private virtualSize = 0
+  private weight = 0
+
+  constructor({ feeRate, network, psbt, witness }: FeeEstimatorOptions) {
+    if (feeRate < 0 || !Number.isSafeInteger(feeRate)) {
+      throw new Error("Invalid feeRate")
+    }
+
+    this.feeRate = +feeRate // convert decimal to whole number that might have passed Number.isSafeInteger check due to precision loss
+    this.network = network
+    this.witness = witness || []
+    this.psbt = psbt || new Psbt({ network: getNetwork(this.network) })
+  }
+
+  private sanityCheckFee() {
+    if (this.fee > MAXIMUM_FEE) {
```

Comment:
Why do we need to throw an error on this? Should we just return the max fee?

Reply:
No. Setting it as the fee needs reverse calculation and would not match the fee rate set by the user. 

This check has been added for the rare edge case where the calculation goes absolutely wrong. fwiw, thats unlikely to happen.